src/main/java/software/amazon/awssdk/s3accessgrants/plugin/S3AccessGrantsAuthSchemeProvider.java

```diff
@@ -93,10 +94,20 @@ public List<AuthSchemeOption> resolveAuthScheme(@NotNull S3AuthSchemeParams auth
             logger.debug(() -> "Access Grants requests will be sent to the region "+destinationRegion);
 
             String S3Prefix;
-            if (operation.equalsIgnoreCase("DELETEOBJECTS")){
+            if (operation.equalsIgnoreCase("DELETEOBJECTS")) {
                 S3Prefix = "s3://" + authSchemeParams.bucket() + getCommonPrefixFromMultiplePrefixes(authSchemeParams.deleteObjectKeys());
             }
-            else{
+            else if (operation.equalsIgnoreCase("COPYOBJECT")) {
+                String[] copySourceArray = authSchemeParams.copySource().split("/", 2);
```

Comment:
Do you think we need a NotNull check for CopySource param here?

Reply:
No we don't. This is done implicitly when copySource is created from sourceBucket and sourceKey.